diff --git a/lib/HUSKYLENS/HUSKYLENS.h b/lib/HUSKYLENS/HUSKYLENS.h
index xxxx..xxxx 100644
--- a/lib/HUSKYLENS/HUSKYLENS.h
+++ b/lib/HUSKYLENS/HUSKYLENS.h
@@ -171,7 +171,7 @@ private:
     bool processReturn(){
         currentIndex = 0;
         if (!wait(COMMAND_RETURN_INFO)) return false;
-        protocolPtr = (Protocol_t*) realloc(protocolPtr, max(protocolInfo.protocolSize, 1) * sizeof(Protocol_t));
+        protocolPtr = (Protocol_t*) realloc(protocolPtr, ((protocolInfo.protocolSize > 1) ? protocolInfo.protocolSize : 1) * sizeof(Protocol_t));
         
         for (int i = 0; i < protocolInfo.protocolSize; i++)
         {
@@ -297,7 +297,7 @@ public:
 
     int available(){
         int result = count();
-        currentIndex = min(currentIndex, result);
+        currentIndex = (currentIndex < result) ? currentIndex : result;
         return result - currentIndex;
     }